be/src/vec/functions/function_string.h

```diff
@@ -356,6 +357,174 @@ class FunctionStrcmp : public IFunction {
+                    unpack_if_const(block.get_by_position(arguments[1]).column);
+
+            const auto* col_str_partition_type_str =
+                    assert_cast<const ColumnString*>(col_partition_type.get());
+
+            std::string to_split_s(chars_list[2]->raw_data(), chars_list[2]->size());
+            std::vector<std::string> str_v;
+            if (to_split_s.size() == input_rows_count * 19) {
+                for (int i = 0; i < to_split_s.size(); i += 19) {
+                    auto vec_res = split(to_split_s.substr(i, 10), "-");
+                    for (const auto& s : vec_res) {
+                        str_v.emplace_back(s);
+                    }
+                }
+            } else {
+                for (int i = 0; i < to_split_s.size(); i += 10) {
+                    auto vec_res = split(to_split_s.substr(i, 10), "-");
+                    for (const auto& s : vec_res) {
+                        str_v.emplace_back(s);
+                    }
+                }
+            }
+            res_data.resize(15 * input_rows_count);
+            for (int i = 0; i < input_rows_count; i++) {
+                int curr_len = 1;
+
+                res_data[res_offset[i - 1]] = 'p';
+                if (compare_str(col_str_partition_type_str->get_chars().raw_data(), "day", 3)) {
+                    for (int j = i * 3; j < i * 3 + 3; j++) {
+                        std::memcpy(&res_data[res_offset[i - 1]] + curr_len, str_v[j].c_str(),
+                                    str_v[j].size());
+                        curr_len += str_v[j].size();
+                    }
+                } else if (compare_str(col_str_partition_type_str->get_chars().raw_data(), "month",
+                                       5)) {
+                    for (int j = i * 3; j < i * 3 + 2; j++) {
+                        std::memcpy(&res_data[res_offset[i - 1]] + curr_len, str_v[j].c_str(),
+                                    str_v[j].size());
+                        curr_len += str_v[j].size();
+                    }
+                    std::memcpy(&res_data[res_offset[i - 1]] + curr_len, "01", 2);
+                    curr_len += 2;
+                } else if (compare_str(col_str_partition_type_str->get_chars().raw_data(), "year",
+                                       4)) {
+                    std::memcpy(&res_data[res_offset[i - 1]] + curr_len, str_v[i * 3].c_str(),
+                                str_v[i * 3].size());
+                    curr_len += str_v[i * 3].size();
+                    std::memcpy(&res_data[res_offset[i - 1]] + curr_len, "0101", 4);
+                    curr_len += 4;
+                } else {
+                    return Status::Error<ErrorCode::INTERNAL_ERROR>(
+                            "the first argument is range , the second argument muse be "
+                            "day/month/year ");
+                }
+                std::memcpy(&res_data[res_offset[i - 1]] + curr_len, "000000", 6);
```

Comment:
do not use `std::memcpy`, just `memcpy`. because we have special optimization on it